src/scene/gsplat/gsplat-compressed-material.js

```diff
@@ -0,0 +1,361 @@
+
+    // early out tiny splats
+    // TODO: figure out length units and expose as uniform parameter
+    // TODO: perhaps make this a shader compile-time option
+    if (dot(v1v2.xy, v1v2.xy) < 4.0 && dot(v1v2.zw, v1v2.zw) < 4.0) {
+        return vec4(0.0, 0.0, 2.0, 1.0);
+    }
+
+    texCoord = vertex_position.xy;
+    color = getColor();
+
+    #ifndef DITHER_NONE
+        id = float(splatId);
+    #endif
+
+    return centerClip + vec4((texCoord.x * v1v2.xy + texCoord.y * v1v2.zw) / viewport * centerClip.w, 0, 0);
+}
+`;
+
+const splatCoreFS = /* glsl */ `
+
+varying vec2 texCoord;
+varying vec4 color;
+
+#ifndef DITHER_NONE
+    varying float id;
+#endif
+
+#ifdef PICK_PASS
+    uniform vec4 uColor;
+#endif
+
+vec4 evalSplat() {
+
+    float A = -dot(texCoord, texCoord);
+    if (A < -4.0) discard;
+    float B = exp(A) * color.a;
+
+    #ifdef PICK_PASS
+        if (B < 0.3) discard;
+        return(uColor);
+    #endif
+
+    #ifndef DITHER_NONE
+        opacityDither(B, id * 0.013);
+    #endif
+
+    #ifdef TONEMAP_ENABLED
+        color.rgb = gammaCorrectOutput(toneMap(decodeGamma(color.rgb)));
+    #endif
+
+    return vec4(color.rgb, B);
+}
+`;
+
+class GSplatCompressedShaderGenerator {
+    generateKey(options) {
+        const vsHash = hashCode(options.vertex);
+        const fsHash = hashCode(options.fragment);
+        return `splat-${options.pass}-${options.gamma}-${options.toneMapping}-${vsHash}-${fsHash}-${options.dither}}`;
+    }
+
+    createShaderDefinition(device, options) {
+
+        const shaderPassInfo = ShaderPass.get(device).getByIndex(options.pass);
+        const shaderPassDefines = shaderPassInfo.shaderDefines;
+
+        const defines =
+            shaderPassDefines +
+            `#define DITHER_${options.dither.toUpperCase()}\n` +
+            `#define TONEMAP_${options.toneMapping === TONEMAP_LINEAR ? 'DISABLED' : 'ENABLED'}\n`;
+
+        const vs = defines + splatCoreVS + options.vertex;
+        const fs = defines + shaderChunks.decodePS +
+            (options.dither === DITHER_NONE ? '' : shaderChunks.bayerPS + shaderChunks.opacityDitherPS) +
+            ShaderGenerator.tonemapCode(options.toneMapping) +
+            ShaderGenerator.gammaCode(options.gamma) +
+            splatCoreFS + options.fragment;
+
+        return ShaderUtils.createDefinition(device, {
+            name: 'SplatShader',
+            attributes: {
+                vertex_position: SEMANTIC_POSITION,
+                vertex_id_attrib: SEMANTIC_ATTR13
+            },
+            vertexCode: vs,
+            fragmentCode: fs
+        });
+    }
+}
+
+const gsplatCompressed = new GSplatCompressedShaderGenerator();
+
+const splatMainVS = `
+    void main(void)
+    {
+        gl_Position = evalSplat();
+    }
+`;
+
+const splatMainFS = `
+    void main(void)
+    {
+        gl_FragColor = evalSplat();
+    }
+`;
+
+/**
+ * @typedef {object} SplatMaterialOptions - The options.
```

Comment:
Did you verify that this symbol doesn't make it to the API reference?

Reply:
yep, it's not exported